### PR TITLE
feat: support new Errored and Bootstrapped events

### PR DESCRIPTION
**Pull request by smira**
Bump embedded etcd to 3.5.6.

Fix the etcd watch API cancelation.

Add controller-runtime conformance tests on top of etcd.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

**Comment by smira** [comment]
See https://github.com/cosi-project/runtime/pull/187

**Review by TimJones — APPROVED** [review]

**Review by DmitriyMV — APPROVED** [review]

File: pkg/state/impl/etcd/etcd.go [hunk 1]
```diff
@@ -376,22 +378,47 @@ func (st *State) Watch(ctx context.Context, resourcePointer resource.Pointer, ch
...
-			case watchResponse = <-watchCh:
+			case watchResponse, ok = <-watchCh:
+				if !ok {
+					// channel closed, quit
+					return
+				}
+			}
+
+			if watchResponse.Err() != nil {
+				select {
```
[the remark this answers is not in the document]
Comment (smira) [reply pkg/state/impl/etcd/etcd.go:413]: yep, makes sense here and in other PRs

[the remark this answers is not in the document]
Comment (DmitriyMV) [reply pkg/state/impl/etcd/etcd.go:413]: Should we add it to github.com/siderolabs/gen?

[the remark this answers is not in the document]
Comment (smira) [reply pkg/state/impl/etcd/etcd.go:413]: probably `SendWithContext`

[the remark this answers is not in the document]
Comment (utkuozdemir) [reply pkg/state/impl/etcd/etcd.go:413]: > I don't like `bool` return there... or the name of the function combined with bool return. it's not clear from the function name what false/true means.

Maybe return a `SendResult` struct with a field like `succeeded` field in it? Or make it just an enum?

**Review by utkuozdemir — APPROVED** [review]

File: pkg/state/impl/etcd/etcd.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (smira) [reply pkg/state/impl/etcd/etcd.go]: I ended up with https://github.com/siderolabs/gen/blob/main/channel/send.go

**Pull request merged**